File: app/src/main/java/com/adisa/diningplus/network/entities/Hall.java
package com.adisa.diningplus.network.entities;

import android.location.Location;

import org.json.JSONException;
import org.json.JSONObject;

public class Hall {
    public String id;
    public String name;
    public String nickname;
    public boolean open;
    public int occupancy;
    public double latitude;
    public double longitude;
    public String address;
    public String phone;

    public double distance;

    public static Hall fromJSON(JSONObject raw) throws JSONException {
        Hall hall = new Hall();
        hall.id = raw.getString("id");
        hall.name = raw.getString("name");
        hall.nickname = raw.getString("nickname");
        hall.open = raw.optBoolean("open");
        hall.occupancy = raw.optInt("occupancy");
        hall.latitude = raw.optDouble("latitude");
        hall.longitude = raw.optDouble("longitude");
        hall.address = raw.optString("address");
        hall.phone = raw.optString("phone");
        return hall;
    }

    public void setDistance(Location location) {
        if (location != null) {
            Location loc = new Location("");
            loc.setLatitude(latitude);
            loc.setLongitude(longitude);
            // Convert m -> km
            distance = location.distanceTo(loc) / 1000;
        }
    }
}

File: app/src/main/java/com/adisa/diningplus/network/entities/Hall.kt
package com.adisa.diningplus.network.entities

import android.location.Location
import org.json.JSONException
import org.json.JSONObject

class Hall(raw: JSONObject) {
    var id: String = raw.getString("id")
    var name: String = raw.getString("name")
    var nickname: String = raw.getString("nickname")
    var open: Boolean = raw.optBoolean("open")
    var occupancy: Int = raw.optInt("occupancy")
    var latitude: Double = raw.optDouble("latitude")
    var longitude: Double = raw.optDouble("longitude")
    var address: String = raw.optString("address")
    var phone: String = raw.optString("phone")

    var distance: Double = 0.0

    fun setDistance(location: Location?) {
        if (location != null) {
            val loc = Location("")
            loc.latitude = latitude
            loc.longitude = longitude
            // Convert m -> km
            distance = location.distanceTo(loc) / 1000.toDouble()
        }
    }
}